packages/drivers/routerlicious-driver/src/documentDeltaConnection.ts

```diff
@@ -107,4 +108,24 @@ export class R11sDocumentDeltaConnection extends DocumentDeltaConnection {
 			url: getUrlForTelemetry(this.url, socketIoPath),
 		};
 	}
+
+	/**
+	 * Disconnect from the websocket
+	 */
+	protected disconnectCore(err?: IAnyDriverError): void {
+		const errorMsg = err?.errorType ?? "Clean disconnect";
+		const isCorruption =
+			err !== undefined &&
+			(err.errorType === FluidErrorTypes.dataCorruptionError ||
+				err.errorType === FluidErrorTypes.dataProcessingError);
+		// tell the server we are disconnecting this client from the document
+		this.socket.emit(
```

Comment:
There are two possible options how we get here:
- It's in response of client deciding to close connection. We likely still have a healthy connection
- In response to connection being closed. (i.e. reacting to disconnect event). In such case connection is no longer healthy.

I'd assume that in second case emitting something on socket is noop, but worth double checking.
Even if works as is, you are not going to get any info in such cases (most of the disconnects recording by client are "transport closed")